Sources/Misc/SandboxEnvironmentDetector.swift

```diff
@@ -30,11 +30,17 @@ final class BundleSandboxEnvironmentDetector: SandboxEnvironmentDetector {
     }
 
     var isSandbox: Bool {
-        guard let url = self.bundle.appStoreReceiptURL else {
+        guard let path = self.bundle.appStoreReceiptURL?.path else {
             return false
         }
 
-        return url.path.contains("sandboxReceipt")
+        // `true` for either `macOS` or `Catalyst`
+        let isMASReceipt = path.contains("MASReceipt/receipt")
+        if isMASReceipt {
+            return path.contains("Xcode/DerivedData")
```

Comment:
OOOHHHH right, great catch! 

could this just be simplified to "if contains sandbox or Xcode/DerivedData"? Feels like we wouldn't get a false positive, right?

Reply:
That would be a false positive for iOS as well. This is an example receipt URL for the iOS simulator:
```
/Users/nachosoto/Library/Developer/Xcode/DerivedData/RevenueCat-areulmnbtfydolgoennnwogndmkr/Build/Products/Debug-iphonesimulator/UnitTests.xctest/base64encodedreceiptsample1.txt
```

So we need to only do that for `macOS`. I think we should hardcode the least amount of strings though, in case this changes in the future, but I believe we can't simplify it more than this?

Reply:
ohhhh simulator, right. I'm so used to testing on device that I forgot 😅